### PR TITLE
MM-11577: Back to previous channel on archive

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Back to previous channel on archive

#### Ticket Link
[MM-11577](https://mattermost.atlassian.net/browse/MM-11577)

#### Device Information
This PR was tested on: [One Plus 5, Android 8.1]

### CODE REVIEW

File: app/actions/views/channel.js [hunk 1]
```diff
@@ -273,6 +274,38 @@ export function selectInitialChannel(teamId) {
...
+        const isGMVisible = lastChannel && lastChannel.type === General.GM_CHANNEL &&
+            isGroupChannelVisible(myPreferences, lastChannel);
+
+        if (
+            myMembers[lastChannelId] &&
+            lastChannel &&
+            (lastChannel.delete_at === 0 || viewArchivedChannels) &&
+            (lastChannel.team_id === teamId || isDMVisible || isGMVisible)
+        ) {
+            handleSelectChannel(lastChannelId)(dispatch, getState);
```
Comment (enahum) [inline app/actions/views/channel.js]: change it to `dispatch(handleSelectChannel(lastChannelId))` same thing with the other one

File: app/screens/channel_info/channel_info.js [hunk 2]
```diff
@@ -207,8 +210,20 @@ export default class ChannelInfo extends PureComponent {
                     if (result.error.server_error_id === 'api.channel.delete_channel.deleted.app_error') {
                         this.props.actions.getChannel(channel.id);
                     }
+                } else if (this.props.viewArchivedChannels) {
+                    this.props.actions.handleSelectChannel(channel.id);
+                    if (Platform.OS === 'android') {
```
Comment (enahum) [inline app/screens/channel_info/channel_info.js]: maybe add a conditional parameter to so it does not emit the event and then replace this block of code with `this.close(false)`

File: app/screens/channel_info/channel_info.js [hunk 3]
```diff
@@ -207,8 +210,20 @@ export default class ChannelInfo extends PureComponent {
...
+                        this.props.navigator.dismissModal({animated: true});
+                    } else {
+                        this.props.navigator.pop({animated: true});
+                    }
                 } else {
-                    this.close();
+                    this.props.actions.selectPenultimateChannel(channel.team_id);
+                    if (Platform.OS === 'android') {
```
Comment (enahum) [inline app/screens/channel_info/channel_info.js]: same here

File: app/screens/channel_info/channel_info.js (hunk 2 above)
Comment (jespino) [reply app/screens/channel_info/channel_info.js]: :+1:

**Comment by jespino** [comment]
It is a bit complicated, I don't have an easy way to mock the fetch. Do you mind to merge it without tests, and I add the tests in another PR?

**Pull request merged**